### PR TITLE
Allow configuring the worker to terminate after all jobs have been processed

### PR DESCRIPTION
**Pull request by Martchus**
See https://progress.opensuse.org/issues/80986

Besides the unit tests I've also conducted 2 full test runs (one with and one without the configuration) and both worked as expected.

### CODE REVIEW

**Review by okurz — CHANGES_REQUESTED** [review]
the code you changed looks fine, I would appreciate the following:

* Extend etc/openqa/workers.ini
* Optionally extend documentation. Not sure if we have a section for possible worker config, feel free to add one
* Put more details in the git commit message to explain the use case

**Comment by Martchus** [comment]
Ok, and see https://github.com/os-autoinst/openQA/pull/3639#issuecomment-744259023 regarding the warning. It is just a deprecation.

**Comment by okurz** [comment]
take a look into https://github.com/os-autoinst/openQA/blob/master/Makefile#L84 for how we create a "no-cleanup" convenience systemd service definition.

File: lib/OpenQA/Worker/Settings.pm [hunk 1]
```diff
@@ -52,6 +52,7 @@ sub new {
 
     # read global settings from environment variables
     $global_settings{LOG_DIR} = $ENV{OPENQA_WORKER_LOGDIR} if $ENV{OPENQA_WORKER_LOGDIR};
+    $global_settings{TERMINATE_AFTER_JOBS_DONE} //= $ENV{OPENQA_WORKER_TERMINATE_AFTER_JOBS_DONE};
```
Comment (kalikiana) [inline lib/OpenQA/Worker/Settings.pm:55]: We should probably have an uncommented example in the default `workers.ini`?

Comment (okurz) [reply lib/OpenQA/Worker/Settings.pm:55]: has been done

**Review by okurz — APPROVED** [review]

**Comment by okurz** [comment]
I have approved but added a label so that you can show "acceptance tests", e.g. from your local env or a staging env.

File: lib/OpenQA/Worker/Settings.pm (hunk 1 above)
Comment (Martchus) [reply lib/OpenQA/Worker/Settings.pm:55]: I have added a comment to `workers.ini`.

**Comment by Martchus** [comment]
As mentioned, I've done 2 tests for both configurations locally. There's nothing special to see in the test results which I could share. What I still want to test is the systemd service file.

**Comment by Martchus** [comment]
I've tested the systemd settings locally and it works: https://w3.suse.de/~mkittler/openQA/worker-log-with-restart.log

**Review by kalikiana — APPROVED** [review]

**Comment by Martchus** [comment]
The package build seems to fail because of the deprecation warning.

**Pull request merged**